### DISCUSSION TITLE
how to use finetuner in a cloud environment

### DISCUSSION THREAD

**Discussion by vahuja4** [jina-ai/finetuner#155] [Q&A]
So, I ran this 

`import finetuner

finetuner.fit(
    model=model,
    interactive=True,
    train_data=data_gen,
    freeze=True,
    to_embedding_model=True,
    input_size=(3, 224, 224),
    output_dim=100
)

`
And, it completed in 20s, but the labeling UI didn't pop up because I am in a cloud based jupyter environment. What is the best way to use a cloud gpu with finetuner?

**Comment by bwanglzu** [comment]
@vahuja4 i see your issue. in notebook/cloud machines we might don't have the `browser` module. We'll see if it is possible to trigger a local browser for that, thanks for reporting this.

My suggestion is, for now, either:
use tailor + tuner api
or:
use it locally.

Sorry for the inconvenience.

**Comment by vahuja4** [comment]
Running it locally (on an M1 Mac) gave me this error: https://ibb.co/nM7TmMT

**Reply by tadejsv** [reply]
@vahuja4 can you give us a small reproducible example so that we can investigate?

**Reply by vahuja4** [reply]
@tadejsv - I just used the celebA example code for my own dataset. No other changes. Used pretrained resnet50 from torchvision.

**Reply by tadejsv** [reply]
What format are images in your dataset?

**Reply by tadejsv** [reply]
If you can, paste the entire python file here - often errors appear in places you don't expect them to

**Reply by bwanglzu** [reply]
and please do verify your data path: `/Users/jina/Downloads/img_align_celeba/*.jpg'`, the data path of the example should be replaced with your local data path.

**Reply by vahuja4** [reply]
Here is the code:
```
from jina.types.document.generators import from_files

def data_gen():
    for d in from_files('/Users/vishal/Downloads/earrings/*.jpg', size=100, to_dataturi=True):
        d.convert_image_datauri_to_blob(color_axis=0)  # `color_axis=-1` for TF/Keras users
        yield d
import torchvision
model = torchvision.models.resnet50(pretrained=True)
import finetuner

finetuner.fit(
    model=model,
    interactive=True,
    train_data=data_gen,
    freeze=True,
    to_embedding_model=True,
    input_size=(3, 224, 224),
    output_dim=100
)

```

**Comment by vahuja4** [comment]
jpg

**Comment by bwanglzu** [comment]
@vahuja4 can you update your jpg, it's now shown up
